akka-actor-typed/src/main/scala/akka/actor/typed/internal/routing/GroupRouterImpl.scala

```diff
@@ -31,6 +33,24 @@ private[akka] final case class GroupRouterBuilder[T] private[akka] (
 
   def withRoundRobinRouting(): GroupRouterBuilder[T] = copy(logicFactory = () => new RoutingLogics.RoundRobinLogic[T])
 
+  override def withConsistentHashingRouting(
+      virtualNodesFactor: Int,
+      mapping: function.Function[T, String],
+      system: ActorSystem[T]): GroupRouterBuilder[T] =
```

Comment:
I think we can avoid the `system` parameter here, if we change the `logicFactory` function in `GroupRouterBuilder`. Something like this:

```scala
private[akka] final case class GroupRouterBuilder[T] private[akka] (
    key: ServiceKey[T],
    logicFactory: ActorSystem[_] => RoutingLogic[T] = _ => new RoutingLogics.RandomLogic[T]())
    extends javadsl.GroupRouter[T]
    with scaladsl.GroupRouter[T] {

  // deferred creation of the actual router
  def apply(ctx: TypedActorContext[T]): Behavior[T] =
    new InitialGroupRouterImpl[T](ctx.asScala, key, logicFactory(ctx.asScala.system))

  def withRandomRouting(): GroupRouterBuilder[T] = copy(logicFactory = _ => new RoutingLogics.RandomLogic[T]())

  def withRoundRobinRouting(): GroupRouterBuilder[T] = copy(logicFactory = _ => new RoutingLogics.RoundRobinLogic[T])

  def withConsistentHashingRouting(
    virtualNodesFactor: Int,
    mapping: T => String): GroupRouterBuilder[T] = {
    import akka.actor.typed.scaladsl.adapter._
    copy(
      logicFactory = system =>
        new RoutingLogics.ConsistentHashingLogic[T](
          virtualNodesFactor,
          mapping,
          system.toClassic.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress))
}
```

Reply:
another improvement is to check what kind of ActorSystem it is before `system.toClassic.asInstanceOf[ExtendedActorSystem]`, because this will not work with the BehaviorTestKit.

We can do that in a separate PR. I think we should add `address: Address` to the typed ActorSystem. That would be useful in many places, instead of converting and casting. Added issue https://github.com/akka/akka/issues/28206 for that.